pkg/apiclient/workflow/workflow.proto

```diff
@@ -182,4 +191,12 @@ service WorkflowService {
     rpc PodLogs (WorkflowLogRequest) returns (stream LogEntry) {
         option (google.api.http).get = "/api/v1/workflows/{namespace}/{name}/{podName}/log";
     }
+
+    rpc SubmitFromResource (WorkflowSubmitRequest) returns (github.com.argoproj.argo.pkg.apis.workflow.v1alpha1.Workflow) {
+        option (google.api.http) = {
+			put: "/api/v1/workflows/{namespace}/{resourceKind}/{resourceName}/submit"
```

Comment:
I think we should have a `submit` endpoint on each of `/api/v1/cron-workflows/`, `/api/v1/workflow-templates/`, `/api/v1/cluster-workflow-templates/` instead of a generalized endpoint for all.

It could look something like:

```
put: "/api/v1/cron-workflows/{namespace}/{resourceName}/submit"
```

Much simpler and make more sense design-wise.

Reply:
Here my comments
 https://github.com/argoproj/argo/issues/2527#issuecomment-604726234
This API is creating Workflow and returning workflow.   It should be similar to CLI `argo submit --from`.  This implementation keeps code wise and testing wise simple and easy. CLI and API Server will have the same implementation

Reply:
Let's discuss with the team today

Reply:
Converted to POST method as per our discussion